exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -173,21 +202,25 @@ func (prwe *prwExporter) PushMetrics(ctx context.Context, md pmetric.Metrics) er
 		return errors.New("shutdown has been called")
 	default:
 
-		tsMap, err := prometheusremotewrite.FromMetrics(md, prwe.exporterSettings)
-		if err != nil {
-			prwe.telemetry.recordTranslationFailure(ctx)
-			prwe.settings.Logger.Debug("failed to translate metrics, exporting remaining metrics", zap.Error(err), zap.Int("translated", len(tsMap)))
+		// If feature flag not enabled support only RW1
+		// TODO remove when feature flag removed
+		if !prwe.enableSendingRW2 {
+			return prwe.pushMetricsV1(ctx, md)
 		}
 
-		prwe.telemetry.recordTranslatedTimeSeries(ctx, len(tsMap))
+		// If feature flag enabled check if we want to send RW1 or RW2
+		switch prwe.RemoteWriteProtoMsg {
+		case config.RemoteWriteProtoMsgV1:
+			// Rw1 case
+			return prwe.pushMetricsV1(ctx, md)
+		case config.RemoteWriteProtoMsgV2:
+			// RW2 case
+			return prwe.pushMetricsV2(ctx, md)
 
-		var m []*prompb.MetricMetadata
-		if prwe.exporterSettings.SendMetadata {
-			m = prometheusremotewrite.OtelMetricsToMetadata(md, prwe.exporterSettings.AddMetricSuffixes)
+		default:
+			return errors.New("unexpected remote-write protobuf message")
```

Comment:
```suggestion
			return fmt.Errorf("unsupported remote-write protobuf message: %v", prwe.RemoteWriteProtoMsg)
```

Reply:
Any special reason to use `fmt.Errorf`? Under the hood, Errorf just calls `errors.New` but do some memory allocation before doing so. We could avoid those allocations here 🤔 

https://cs.opensource.google/go/go/+/master:src/fmt/errors.go;l=22-30